src/gui/resourcetreeitem.h

```diff
@@ -6,76 +6,88 @@
-    QList<ResourceTreeItem*> _children;
-    QString _name = "DEFAULTNAME";
-    ResourceTreeItem *_parent = nullptr;
-    QString _fullPath = "DEFAULTPATH";
-    bool _isDir;
-    Source _source;
-    Aurora::FileType _fileType;
-    Aurora::ResourceType _resourceType;
-    qint64 _size;
-
-    mutable bool _triedDuration;
-    mutable uint64 _duration;
-
-    /** Internal resource information. */
 public:
-    struct Data {
-        QString path;
+    enum Source {
+        kSourceDirectory= 0,
+        kSourceFile = 1,
+        kSourceArchive = 2,
+        kSourceArchiveFile = 3
+    };
 
-        Aurora::Archive *archive = nullptr;
-        bool addedArchiveMembers = false;
+    struct ArchiveInfo {
+        Aurora::Archive *archive;
+        bool addedArchiveMembers;
         uint32 archiveIndex;
     };
 
-private:
-    Data _data;
+    struct FileInfo {
+        QString fileName;
+        QString fullPath;
+        bool isDir;
+        Source source;
+        Aurora::FileType fileType;
+        Aurora::ResourceType resourceType;
+        qint64 size;
+    };
 
-public:
-    ResourceTreeItem(const QString fullPath, Aurora::Archive *archive, Aurora::Archive::Resource &resource, ResourceTreeItem *parent);
+    struct SoundInfo {
+        mutable bool triedDuration;
+        mutable uint64 duration;
+    };
+
+
+    ResourceTreeItem(Aurora::Archive *archive, Aurora::Archive::Resource &resource, ResourceTreeItem *parent);
     ResourceTreeItem(const QString fullPath, ResourceTreeItem *parent);
     ~ResourceTreeItem();
 
-    void addChild(ResourceTreeItem *child);
+    /** Model structure. **/
+    bool hasChildren();
     bool insertChild(int position, ResourceTreeItem *child);
-    void setParent(ResourceTreeItem *parent);
     int childCount();
-    ResourceTreeItem *child(int row);
     int row();
+    ResourceTreeItem *childAt(int row);
+    ResourceTreeItem *parent();
+    void appendChild(ResourceTreeItem *child);
+    void setParent(ResourceTreeItem *parent);
+
+    /** File info. **/
+    Aurora::FileType getFileType();
+    Aurora::ResourceType getResourceType();
     bool isDir();
+    QFileInfo getFileInfo();
+    qint64 getSize();
     QString &getPath();
-    ResourceTreeItem *getParent();
     QString getName();
-    void setFileInfo(const QFileInfo info);
-    QFileInfo getFileInfo();
     Source getSource();
-    Aurora::FileType getFileType();
-    Aurora::ResourceType getResourceType();
-    qint64 getSize();
-    bool hasChildren();
+
+    /** Resource information. **/
+    ArchiveInfo &getData();
+    Common::SeekableReadStream *getResourceData() const;
     Images::Decoder *getImage();
     Images::Decoder *getImage(Common::SeekableReadStream &res, Aurora::FileType type);
-    Common::SeekableReadStream *getResourceData() const;
-    Data &getData();
-    uint64 getSoundDuration() const;
     Sound::AudioStream *getAudioStream() const;
+    uint64 getSoundDuration() const;
+
+private:
+    /** Model information. **/
```

Comment:
`/** **/` denotes a [Doxygen](https://xoreos.org/doxygen/)-comment. Those don't span multiple items, only the one immediately following the comment. In this case, the `_parent` member variable.

Here, you probably want to use normal comments instead.